synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go

```diff
@@ -339,7 +356,12 @@ func checkProcessBatchResultMatchExpected(data *ProcessData, processBatchResp *s
 	var err error = nil
 	var trustedBatch = data.TrustedBatch
 	if trustedBatch == nil {
-		panic("trustedBatch is nil")
+		log.Error("trustedBatch is nil, it never should be nil")
+		return fmt.Errorf("trustedBatch is nil, it never should be nil")
```

Comment:
just to avoid duplicated string, we could replace it by: 
```go
err := fmt.Errorf("trustedBatch is nil, it never should be nil")
log.Error(err.Error())
return err
```